File: Freestylecoding.Math.FSharp.Tests/Natural/AdditiveIdentity.fs
namespace Natural

open Xunit
open Freestylecoding.Math

type public AdditiveIdentity() =
    static member GetAdditiveIdentity( this:'T when 'T :> System.Numerics.IAdditiveIdentity<'T,'T> ) : 'T =
        'T.AdditiveIdentity

    [<Fact>]
    member public this.IsIdentityCorrect () =
        Assert.Equal(
            Natural.Zero,
            AdditiveIdentity.GetAdditiveIdentity( Natural.Unit )
        )

File: Freestylecoding.Math.CSharp.Tests/Natural/AdditiveIdentity.cs
using Xunit;

namespace Natural.Tests;

using static Freestylecoding.Math.CSharp.Tests.Helpers;
using Natural = Freestylecoding.Math.Natural;

public class AdditiveIdentity {
	[Fact]
	public void IsIdentityCorrect() =>
		Assert.Equal(
			Natural.Zero,
			Natural.AdditiveIdentity
		);
}